src/Makefile

```diff
@@ -323,13 +323,27 @@ clean-support:
 
 cleanall: clean clean-flisp clean-support
 
-analyzegc:
+$(build_shlibdir)/libGCCheckerPlugin.$(SHLIB_EXT): $(SRCDIR)/clangsa/GCChecker.cpp
+	@$(call PRINT_CC, $(CXX) -g -shared -fno-rtti -fPIC -o $@ -std=c++11 -DCLANG_PLUGIN -I$(build_includedir)  -I$(LLVM_SRC_DIR)/include -L$(build_libdir) $< -lclangAnalysis -lclangStaticAnalyzerCore -lclangASTMatchers -lclangAST -lclangLex -lclangBasic)
+
+define CLANG_ANALYZE
+clang-sa-$(1): $$(build_shlibdir)/libGCCheckerPlugin.$$(SHLIB_EXT)
+	@$$(call PRINT_ANALYZE, $$(build_depsbindir)/clang --analyze -Xanalyzer -analyzer-output=text -Xclang -load -Xclang $$(build_shlibdir)/libGCCheckerPlugin.$$(SHLIB_EXT) $$(CPPFLAGS) $$(CFLAGS) $$(DEBUGFLAGS) -Xclang -analyzer-checker=core$$(COMMA)julia.GCChecker --analyzer-no-default-checks -fcolor-diagnostics -Werror -x c $$(SRCDIR)/$(1).c)
+endef
+
+# Build a Makefile target for each file we want to analyze
+$(foreach S,$(RUNTIME_C_SRCS),$(eval $(call CLANG_ANALYZE,$(S))))
+
+# Note that for a default install, you will need to have run the following
+# before attempting this static analysis, so that all necessary headers
+# and dependencies are properly installed:
+#   make -c deps install-libuv install-utf8proc install-unwind
+analyzegc: $(BUILDDIR)/julia_version.h $(BUILDDIR)/julia_flisp.boot.inc $(build_shlibdir)/libGCCheckerPlugin.$(SHLIB_EXT)
 ifeq ($(USE_BINARYBUILDER_LLVM),0)
 ifneq ($(BUILD_LLVM_CLANG),1)
 	$(error Clang must be available to use the clang analyzer. Either build it (BUILD_LLVM_CLANG) or use BinaryBuilder)
 endif
 endif
-	$(MAKE) -C clangsa
-	$(build_depsbindir)/clang --analyze -Xanalyzer -analyzer-output=text -Xclang -load -Xclang $(build_shlibdir)/libGCCheckerPlugin.$(SHLIB_EXT)  $(CPPFLAGS) $(CFLAGS) $(DEBUGFLAGS) -Xclang -analyzer-checker=core,julia.GCChecker --analyzer-no-default-checks -fcolor-diagnostics -Werror -x c $(RUNTIME_C_SRCS:%=$(SRCDIR)/%.c)
+	$(MAKE) $(addprefix clang-sa-,$(RUNTIME_C_SRCS))
```

Comment:
recursive calls to the same Makefile tend to be a bad idea

Reply:
Agreed; the need for this has decreased significantly now that I've eliminated the rest of the calls within this rule; I'll fix this to instead depend on a `clang-sa-deps-check` rule that conditionally runs the `$(error)`, then we can do everything within the same invocation.